Lib/distutils/sysconfig.py

```diff
@@ -232,7 +234,8 @@ def customize_compiler(compiler):
             compiler_cxx=cxx,
             linker_so=ldshared,
             linker_exe=cc,
-            archiver=archiver)
+            archiver=archiver,
+            ranlib=ranlib)
```

Comment:
`ranlib` should be added here only if it is in `compiler.executables`. Otherwise this would be adding ranlib to other OS as well.

Reply:
Hey, thanks for the feedback. I may be misunderstanding something but this change would add ranlib only if it was explicitly passed as an environment variable (or set in the Python Makefile), right? If I am misunderstanding something I would love if you could sketch the code change since I know next to nothing about the distutils codebase (apart from what I have learned debugging this issue).

I think this change is really OS independent – it's just that it is easier to trigger the bug on macOS since it breaks many Linux-centric assumptions.